fus.c

```diff
@@ -428,18 +429,74 @@ filelist_loadcb (Pool     *pool,
   return 1;
 }
 
+static gboolean
+download_to_path (const char   *url,
+                  const char   *path,
+                  SoupSession  *session,
+                  gboolean      force_download,
+                  GError      **error)
+{
+  g_autoptr(SoupURI) parsed = soup_uri_new (url);
+  if (!SOUP_URI_VALID_FOR_HTTP (parsed))
+    {
+      g_debug ("%s is already a local file", url);
+      return TRUE;
+    }
+
+  if (g_file_test (path, G_FILE_TEST_IS_REGULAR) && !force_download)
+    return TRUE;
+
+  g_debug ("Downloading %s to %s", url, path);
+
+  g_autoptr(SoupMessage) msg = soup_message_new_from_uri ("GET", parsed);
+  g_autoptr(GInputStream) istream = soup_session_send (session, msg, NULL, error);
+  if (!SOUP_STATUS_IS_SUCCESSFUL (msg->status_code))
+    return FALSE;
+
+  g_autoptr(GFile) file = g_file_new_for_path (path);
+  g_autoptr(GFileOutputStream) ostream = g_file_replace (file,
+                                                         NULL,
+                                                         FALSE,
+                                                         G_FILE_CREATE_REPLACE_DESTINATION,
+                                                         NULL,
+                                                         error);
+  if (!ostream)
+    return FALSE;
+
+  if (g_output_stream_splice (G_OUTPUT_STREAM (ostream), istream, 0, NULL, error) == -1)
+    return FALSE;
+
+  return TRUE;
+}
+
 static Repo *
-create_repo (Pool        *pool,
-             const char  *name,
-             const char  *path,
-             GError     **error)
+create_repo (Pool         *pool,
+             const char   *name,
+             const char   *path,
+             SoupSession  *session,
+             GError      **error)
 {
   FILE *fp;
   Id chksumtype;
   const unsigned char *chksum;
   const char *fname;
 
-  fp = solv_xfopen (pool_tmpjoin (pool, path, "/", "repodata/repomd.xml"), "r");
+  g_autofree gchar *destdir = g_build_filename (g_get_user_cache_dir (),
+                                                "fus", "repodata", name,
```

Comment:
isn't it /fus/name/repodata?

Reply:
Yes, I'll fix it.

Reply:
It's `/fus/name/repodata/` now with the fixup.